optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2022 The HuggingFace and Meta Team.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+
+import torch
+
+from .models import is_module_fast
+from .utils import check_if_pytorch_greater_112, is_accelerate_available
+
+
+if is_accelerate_available():
+    from accelerate import dispatch_model, infer_auto_device_map
+    from accelerate.hooks import remove_hook_from_module
+    from accelerate.utils import named_module_tensors, set_module_tensor_to_device
+
+
+def init_accelerate_hook(module):
+    r"""
+    This function initializes `accelerate` hooks by setting the
+    parameters of the module on the `cpu` if there is any offloading involved
+
+    Args:
+        `module` (`torch.nn.Module`, **required**):
+            The input module to initialize
+    Returns:
+        The initialized module
+    """
+    for name, child in module.named_children():
+        if hasattr(child, "_hf_hook"):
+            if child._hf_hook.weights_map:
+                hook = child._hf_hook
+                if child._hf_hook.offload:
+                    for name, _ in named_module_tensors(
+                        child, include_buffers=hook.offload_buffers, recurse=hook.place_submodules
+                    ):
+                        set_module_tensor_to_device(child, name, "cpu", value=hook.weights_map[name])
```

Comment:
The offloading code puts the tensor on the meta device, as the goal is to avoid using RAM. Accelerate keeps a copy of the state dict which is disconnected form the model params, so you will end up with twice the model in RAM by doing this during a period of time.

Reply:
I see! Ideally I want the parameters to be set on the `cpu` before the conversion happens. I believe there is a way to remove the copy of the state dict from the RAM when setting the param on the `cpu`. Will investigate that!

Reply:
Fixed in 4fcd0b3f89c1da164c56005e69066076d4508ea4